tests/BearSerializationTest.py

```diff
@@ -0,0 +1,34 @@
+import itertools
+import json
+import os
+import json
+import unittest
+
+from pyprint.NullPrinter import NullPrinter
+
+from coalib.bears.BEAR_KIND import BEAR_KIND
+from coalib.collecting.Collectors import collect_bears
+from coalib.output.JSONEncoder import create_json_encoder
+from coalib.output.printers.LogPrinter import LogPrinter
+
+
+class BearSerializationTest(unittest.TestCase):
+    """
+    Test to make sure all bears are serializable.
+    """
+
+    def setUp(self):
+        printer = LogPrinter(NullPrinter())
+        local_bears, global_bears = collect_bears(
+            [os.path.abspath("bears")],
+            ["**"],
+            [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL],
+            printer,
+            warn_if_unused_glob=False)
+        self.bears = list(itertools.chain(local_bears, global_bears))
+        self.JSONEncoder = create_json_encoder(use_relpath=False)
+
+    def test(self):
+        json.dumps(self.bears, cls=self.JSONEncoder)
+        # If no exception is thrown then the test passes
```

Comment:
Um I am not on PC now. Um alright. Let's at least catch the exception, log
the error neatly and then raise assertion error. :D
On 12 Jul 2016 6:49 pm, "Sheikh Araf" <notifications@github.com> wrote:

> In tests/BearSerializationTest.py
> <https://github.com/coala-analyzer/coala-bears/pull/574#discussion_r70435612>
> :
>
> > +    """
> > +
> > +    def setUp(self):
> > +        printer = LogPrinter(NullPrinter())
> > +        local_bears, global_bears = collect_bears(
> > +            [os.path.abspath("bears")],
> > +            ["**"],
> > +            [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL],
> > +            printer,
> > +            warn_if_unused_glob=False)
> > +        self.bears = list(itertools.chain(local_bears, global_bears))
> > +        self.JSONEncoder = create_json_encoder(use_relpath=False)
> > +
> > +    def test(self):
> > +        json.dumps(self.bears, cls=self.JSONEncoder)
> > +        # If no exception is thrown then the test passes
>
> This is exactly why I didn't go with catching exceptions. I'm not sure how
> many exceptions can be thrown.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/574/files/5818022c8a7d2e6d652f0a63d36a6b343621728c#r70435612>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOef0JrQHmR5yHqFBiXJKQIl1JOmIks5qU5RXgaJpZM4JIza0>
> .
>